src/games/strategy/triplea/ui/MapData.java

```diff
@@ -700,6 +701,50 @@ private Rectangle getBoundingRectWithTranslate(final List<Polygon> polys, final
   }
 
   /**
+   * Find the best rectangle inside the territory to place the name in. Finds the rectangle
+   * that can fit the name, that is the closest to the vertical center, and has a large width at
+   * that location. If there isn't any rectangles that can fit the name then default back to the
+   * bounding rectangle.
+   */
+  public Rectangle getBestTerritoryNameRect(final Territory t, final FontMetrics fm) {
+    final Rectangle territoryBounds = getBoundingRect(t);
+    Rectangle maxRectangle = territoryBounds;
+    final int minX = territoryBounds.x;
```

Comment:
Please inline `minX` and `minY`.
There are too many variables declared to be remembered at once. This is bad for being able to understand things easily when you have to look up everything because your short term memory is out of space. There are 11 variables declared.. The only way to remember them is to figure out which ones can be ignored. This is work that should be done for the reader.